src/cockpit/channels/dbus.py

```diff
@@ -254,15 +258,34 @@ async def get_ready():
         else:
             self.ready()
 
-    def add_match(self, rule, handler):
+    def add_signal_handler(self, handler, **kwargs):
+        r = dict(**kwargs)
+        r['type'] = 'signal'
+        if 'sender' not in r and self.name is not None:
+            r['sender'] = self.name
+        # HACK - https://github.com/bus1/dbus-broker/issues/309
+        # path_namespace='/' in a rule does not work.
+        if r.get('path_namespace') == "/":
+            del r['path_namespace']
+        def filter_owner(message):
+            if self.owner is None or self.owner == message.get_sender():
```

Comment:
is that right?  In case the name has no owner I'd rather think we want to drop the message...

Reply:
Yes, I think you are right.  We don't want to filter at all when `self.name is None`, but this code only runs when `self.name is not None` and then we should insist on knowing the owner.

I guess NameOwnerChanged will be received before any signal from the new owner, right?